dff/messengers/telegram.py

```diff
@@ -0,0 +1,179 @@
+from telegram.ext.filters import ALL
+
+from dff.messengers.common import MessengerInterface
+from dff.pipeline import Pipeline
+from dff.pipeline.types import PipelineRunnerFunction
+from dff.script.core.context import Context
+from dff.script.core.message import Animation, Audio, Button, CallbackQuery, Contact, DataAttachment, Document, Image, Invoice, Keyboard, Location, Message, Poll, PollOption, Video
+
+
+class _AbstractTelegramInterface(MessengerInterface):  # pragma: no cover
+    def __init__(self, token: str) -> None:
+        self.application = Application.builder().token(token).build()
+        self.application.add_handler(MessageHandler(ALL, self.on_message))
+        self.application.add_handler(CallbackQueryHandler(self.on_callback))
+
+    async def populate_attachment(self, attachment: DataAttachment) -> None:  # pragma: no cover
+        if attachment.title is not None and attachment.id is not None:
+            file_name = Path(gettempdir()) / str(attachment.title)
+            if not file_name.exists():
+                await (await self.application.bot.get_file(attachment.id)).download_to_drive(file_name)
+            attachment.source = FilePath(file_name)
+        else:
+            raise ValueError(f"For attachment {attachment} title or id is not defined!")
+
+    async def extract_message_from_telegram(self, update: TelegramMessage) -> Message:  # pragma: no cover
+        message = Message()
+        message.attachments = list()
+
+        if update.text is not None:
+            message.text = update.text
+        if update.location is not None:
+            message.attachments += [Location(latitude=update.location.latitude, longitude=update.location.longitude)]
+        if update.contact is not None:
+            message.attachments += [Contact(phone_number=update.contact.phone_number, first_name=update.contact.first_name, last_name=update.contact.last_name)]
+        if update.invoice is not None:
+            message.attachments += [Invoice(title=update.invoice.title, description=update.invoice.description, currency=update.invoice.currency, amount=update.invoice.total_amount)]
+        if update.poll is not None:
+            message.attachments += [Poll(question=update.poll.question, options=[PollOption(text=option.text, votes=option.voter_count) for option in update.poll.options])]
+        if update.audio is not None:
+            message.attachments += [Audio(id=update.audio.file_id, title=update.audio.file_unique_id)]
+        if update.video is not None:
+            message.attachments += [Video(id=update.video.file_id, title=update.video.file_unique_id)]
+        if update.animation is not None:
+            message.attachments += [Animation(id=update.animation.file_id, title=update.animation.file_unique_id)]
+        if len(update.photo) > 0:
+            message.attachments += [Image(id=picture.file_id, title=picture.file_unique_id) for picture in update.photo]
+        if update.document is not None:
```

Comment:
Isn't it possible to send multiple documents as well?
Or are they treated as several updates?

Reply:
Could you check what happens if user sends two files?
I think that it would be split into two updates. We might want to consider merging them.

Reply:
Well, this appeared to be easier than I expected: I found out that telegram doesn't support sending multiple documents in one message at all (at least from mobile device) - or at least I didn't find a way to do that :(